priam/src/main/java/com/netflix/priam/notification/AWSSnsNotificationService.java

```diff
@@ -0,0 +1,121 @@
+package com.netflix.priam.notification;
+
+import com.amazonaws.services.sns.AmazonSNS;
+import com.netflix.priam.IConfiguration;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.sns.AmazonSNSClient;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.netflix.priam.aws.IAMCredential;
+import com.netflix.priam.merics.IMeasurement;
+import com.netflix.priam.merics.IMetricPublisher;
+import com.netflix.priam.utils.BoundedExponentialRetryCallable;
+
+/*
+ * A single, persisted, connection to Amazon SNS.
+ */
+@Singleton
+public class AWSSnsNotificationService implements INotificationService {
+	private static final Logger logger = LoggerFactory.getLogger(AWSSnsNotificationService.class);
+
+	private IConfiguration configuration;
+	private AmazonSNS snsClient;
+	private IMetricPublisher metricPublisher;
+	private IMeasurement<Object> notificationMeasurement;
+	
+	@Inject
+	public AWSSnsNotificationService(IConfiguration config, IAMCredential iamCredential
```

Comment:
Since this is the default implementation now, do we have a good Noop for IAMCredentials? Looks like metricspublisher has a noop and config is required.

Reply:
IAM Credentials are used for every AWS operations and thus is always provided for Priam to work